theme/assets/sass/components/_404.scss

```
@@ -0,0 +1,11 @@
+/* stylelint-disable selector-class-pattern */
+.Tree-items {
+    .Tree-item {
+        // Hide 404 page from sidebar in production, since it has a different URL after Webpack build
+        a[href='/docs/404'],
```

untested, but the [*=](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors) should select any `a` elements with `href` attribute containing `404` so it could simplify the selector list here
```suggestion
        a[href*='404'],
```

---

TIL :)